homeassistant/components/godice/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""The godice integration."""
+
+from __future__ import annotations
+
+import logging
+
+import bleak
+import godice
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import DOMAIN
+
+PLATFORMS = [Platform.SENSOR]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up godice from a config entry."""
+    _LOGGER.debug("Setup started")
+
+    def on_disconnect_callback(_ble_data):
+        _LOGGER.debug("on_disconnect_callback called")
+        is_disconnected_by_request = hass.data[DOMAIN][entry.entry_id][
+            "is_disconnected_by_request"
+        ]
+        if not is_disconnected_by_request:
+            hass.create_task(hass.config_entries.async_reload(entry.entry_id))
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, entry.data["address"])
+    assert ble_device is not None
+    client = bleak.BleakClient(
+        ble_device, timeout=20, disconnected_callback=on_disconnect_callback
+    )
+    dice = godice.create(client, godice.Shell.D6)
+    try:
+        await dice.connect()
+        await dice.pulse_led(
+            pulse_count=2, on_time_ms=50, off_time_ms=20, rgb_tuple=(0, 255, 0)
+        )
+    except Exception as err:
+        raise ConfigEntryNotReady("Device not found") from err
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "device": dice,
+        "device_info": create_devinfo(entry),
+        "is_disconnected_by_request": False,
```

Comment:
Here you can use `CONF_DEVICE` from homeassistant.const, for the device_info and is_disconnect_by_request I would create your own constants in const.py like CONF_DEVICE_INFO = "device_info" and use that everywhere